### PR TITLE
Added usefull functions (getallvm,start,stop,...)

### PR DESCRIPTION
**Pull request by lucapiccio**
Added usefull functions (get_vms,start_vm,shutdown_vm,stop_vm,resume_vm,suspend_vm,clone_vm,snapshot_vm)

This are some function to simplify using the api

### CODE REVIEW

File: pve2_api.class.php [hunk 1]
```diff
@@ -320,6 +320,229 @@ public function get_next_vmid () {
...
+                    $key_values = array_column($vms_list, 'vmid'); 
+                    array_multisort($key_values, SORT_ASC, $vms_list);
+                    foreach($vms_list as &$row) {
+                        $row[node] = $node_name;
+                    }
+                    $result = array_merge($result, $vms_list);
+    	        } 
+        		if (count($result) > 0) {
+        		    $this->$cluster_vms_list = $result;
+                    return $this->$cluster_vms_list;
```
Comment (CpuID) [inline pve2_api.class.php:343]: spaces vs tabs (inconsistency)

File: pve2_api.class.php [hunk 2]
```diff
@@ -320,6 +320,229 @@ public function get_next_vmid () {
...
+        		} else {
+        			error_log(" Empty list of vms returned in this cluster.");
+        			return false;
+        		}
+    	    }
```
Comment (CpuID) [inline pve2_api.class.php:348]: spaces vs tabs

**Review by CpuID — CHANGES_REQUESTED** [review]
code looks fine, I haven't tested it but assuming it works, no major objections

if you can fix the spaces vs tabs inconsistencies I'll approve it, thx

File: pve2_api.class.php (hunk 1 above)
Comment (lucapiccio) [reply pve2_api.class.php]: Done

File: pve2_api.class.php (hunk 2 above)
Comment (lucapiccio) [reply pve2_api.class.php]: Done

**Pull request merged**